xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -198,6 +221,9 @@ viewers.Comments = Class.create({
       // Add listener for submit
       form.down("input[type='submit']").observe('click', function(event) {
         event.stop();
+
+        // triggers the copy of the content of the rich editor to hidden fields
+        jQuery(document).trigger('xwiki:actions:beforeSave');
```

Comment:
This mix of jQuery and Prototype.js is painful to read.. For small changes, like triggering an event, I don't see why you would switch to jQuery in the middle of Prototype.js code. For large changes, we need to extract the new jQuery based code and place it at the end, after the Prototype.js code, so that it's clear how much code we still need to convert.